### DISCUSSION TITLE
babel과 webpack의 기본 개념을 설명해 보세요.

### DISCUSSION THREAD

**Discussion by bona373737** [wanted-9th-3team/frontend-study#26] [프론트 기술 질문 공간] [answered]
babel,webpack

**Comment by WooGie911** [comment, marked as answer]
웹팩(Webpack)은 오픈 소스 자바스크립트 모듈 번들러입니다.

많은 파일들을 다운받아와야 할 때 네트워크 부하가 커지고 느려지며 
이름이 같은 변수나 함수로 충돌 가능성이 있습니다. 

이를 해결하기 위해 여러개의 파일을 하나의 파일로 모아주는 번들러를 사용하는데 
웹팩이 그 역할을 합니다.


바벨(Babel)은 대표적인 트랜스 파일러로 
특정 언어로 작성된 코드를 비슷한 다른 언어로 변환 시켜 줍니다. 

언어는 빠르고 끊임없이 발전하고 있기 때문에 
모든 브라우저가 최신 언어를 사용하고 있지 않습니다. 

따라서 최신 기능을 제공하고 있는 최신 언어를 구 언어로 변환하여 사용함으로써 
최신 언어에만 있는 기능을 구 언어를 사용하는 브라우저에서 사용할 수 있습니다.

**Comment by michoball** [comment]
웹팩은 여러개의 파일을 하나로 합쳐주는 번들러 입니다.  여러 파일로 나눠져있는 js 파일을 하나로 통합하고 압축하고 최적화해줘서 네트워크 비용을 줄여줍니다. 

바벨은 대부분의 브라우져가 이해할 수 있는 문법으로 개발자가 작성한 js 코드를 변환하여주는 컴파일러 중 하나입니다. 최신 js 문법을 아직 지원하지 않는 브라우져를 위해서 해당 문법을 예전 예를들면 es5 문법으로 바꿔주는 일을 합니다. 

만약 예전에 존재하는 문법으로 대체가 불가능한 경우에는 이전 문법을 이용해서 다시 구현해주 툴이 있는데 이것은 polyfill이라고 합니다.

**Comment by skyhanull** [comment]
웹팩은 웹에서 사용되는 모든 자원(assets)을 번들링 해주는 도구입니다. 번들링의 개념은 여러개의 파일 중에 종속성이 존재하는 파일을 하나의 파일로 묶어 패키징을 시키는 과정을 의미합니다.
 바벨(babel)은 트랜스파일러(transpiler)로, 모던 자바스크립트 코드를 '구 표준'을 준수하는 코드로 변환해주는 것을 말합니다.

**Comment by BaekYuri** [comment]
웹팩은 자바스크립트를 위한 정적 모듈 번들러 입니다. 모듈 단위로 개발할 수 있기 때문에 유지보수성을 높일 수 있습니다.
바벨은 ES6 이상의 자바스크립트를 구형 브라우저에서도 동작할 수 있도록 ES5 이하의 코드로 트랜스파일링 해줍니다.

**Comment by bona373737** [comment]
webpack : 여러개로 나눠진 자바스크립트 파일을 html이 실행할 수 있는 하나의 자바스크립트 파일로 합쳐주는 모듈 번들러.
번들러를 사용하면 여러개 파일을 하나로 묶어주기 때문에 네트워크 접속의 부담을 줄여 더 빠른 서비스를 제공할 수 있다.

babel :  Javascript Compiler.
브라우저들이 최신 Javscript 코드를 읽지 못하는 것을 위해 최신 버젼의 Javascript를 브라우저가 읽을 수 있도록 옛날 버젼의 Javascript 코드로 변환해주는 역할을 한다.
